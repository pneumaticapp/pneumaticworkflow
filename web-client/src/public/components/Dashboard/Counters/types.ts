/* eslint-disable */
/* prettier-ignore */
export enum EDashboardCounterType {
  Started = 'started',
  InProgress = 'in-progress',
  Overdue = 'overdue',
  Completed = 'completed',
}
