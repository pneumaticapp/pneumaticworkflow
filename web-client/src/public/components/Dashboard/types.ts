export type TDashboardColor = 'blue' | 'yellow' | 'green' | 'red' | 'gray';
