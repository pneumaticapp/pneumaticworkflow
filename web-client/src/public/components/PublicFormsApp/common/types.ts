/* eslint-disable */
/* prettier-ignore */
import { IKickoff } from '../../../types/template';

export interface IPublicForm {
  accountId: number;
  name: string;
  description: string;
  kickoff: IKickoff;
  showCaptcha: boolean;
}

export enum EPublicFormState {
  Loading = 'loading',
  FormNotFound = 'form-not-found',
  WaitingForAction = 'waiting-for-action',
  Submitting = 'submitting',
  Submitted = 'submitted',
}
