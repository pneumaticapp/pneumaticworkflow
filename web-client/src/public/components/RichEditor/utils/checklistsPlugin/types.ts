export type TChecklistItemData = {
  listApiName: string;
  itemApiName: string;
};
