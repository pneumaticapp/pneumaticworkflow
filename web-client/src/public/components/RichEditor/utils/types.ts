export enum ECustomEditorEntities {
  File = 'file',
  Image = 'image',
  Video = 'video',
  Variable = 'variable',
  Mention = 'mention',
  Link = 'LINK',
}

export const CUSTOM_EDITOR_ENTITES = Object.values(ECustomEditorEntities);

export type TEditorAttachment = {
  url: string;
  id?: number;
  name?: string;
};

export enum EEditorBlock {
  Unstyled  = 'unstyled',
  HeaderOne = 'header-one',
  HeaderTwo = 'header-two',
  HeaderThree = 'header-three',
  HeaderFour = 'header-four',
  HeaderFive = 'header-five',
  HeaderSix = 'header-six',
  UnorderedListItem = 'unordered-list-item',
  OrderedListItem = 'ordered-list-item',
  Blockquote = 'blockquote',
  Code = 'code-block',
  Atomic = 'atomic',
};

export enum EEditorStyle {
  Bold = 'BOLD',
  Italic = 'ITALIC',
};
