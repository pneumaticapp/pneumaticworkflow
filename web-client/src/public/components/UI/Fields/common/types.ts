export type TForegroundColor = 'white' | 'beige';
