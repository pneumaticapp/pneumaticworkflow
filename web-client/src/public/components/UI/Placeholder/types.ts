/* eslint-disable */
/* prettier-ignore */
export type TPlaceholderMood = 'good' | 'neutral' | 'bad';
export interface IPlaceholderIconProps extends React.SVGAttributes<SVGElement> {
  mood: TPlaceholderMood;
}
