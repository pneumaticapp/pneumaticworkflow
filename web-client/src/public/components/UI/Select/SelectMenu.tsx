import * as React from 'react';
import classnames from 'classnames';
import { DropdownItem, DropdownMenu, DropdownToggle, Dropdown } from 'reactstrap';

import { ExpandIcon } from '../../icons';
import { IntlMessages } from '../../IntlMessages';

import styles from './Select.css';

export interface ISelectMenuProps<T extends string> {
  activeValue: T;
  values: T[];
  toggleClassName?: string;
  toggleTextClassName?: string;
  arrowClassName?: string;
  menuClassName?: string;
  containerClassName?: string;
  isDisabled?: boolean;
  hideSelectedOption?: boolean;
  onChange(value: T): void;
  Icon?(props: React.SVGAttributes<SVGElement>): JSX.Element;
}

export const SelectMenu = <T extends string>({
  toggleClassName,
  isDisabled,
  values,
  activeValue,
  toggleTextClassName,
  arrowClassName,
  menuClassName,
  hideSelectedOption,
  containerClassName,
  onChange,
  Icon,
}: ISelectMenuProps<T>) => {
  const [isDropdownOpen, setIsDropdownOpen] = React.useState(false);
  const getIntlId = (value: T) => `sorting.${value}`;

  const handleClickItem = (value: T) => () => {
    if (value !== activeValue) {
      onChange(value);
    }
  };

  const handleToggleDropdown = () => {
    setIsDropdownOpen(!isDropdownOpen);
  };

  return (
    <Dropdown
      className={classnames(styles['container'], 'dropdown-menu-right dropdown', containerClassName)}
      isOpen={isDropdownOpen}
      setActiveFromChild={isDropdownOpen}
      toggle={handleToggleDropdown}
    >
      <DropdownToggle
        tag="button"
        className={classnames(styles['active-value'], toggleClassName, isDisabled && styles['active-value_disabled'])}
      >
        {Icon && <Icon className={styles['icon']} />}
        <IntlMessages id={getIntlId(activeValue)}>
          {(text) => <span className={classnames(styles['active-value__text'], toggleTextClassName)}>{text}</span>}
        </IntlMessages>

        <ExpandIcon className={classnames(styles['expand-icon'], arrowClassName)} />
      </DropdownToggle>
      <DropdownMenu className={classnames(styles['dropdown-menu'], menuClassName)}>
        {(values as T[]).map((value) => {
          if (hideSelectedOption && value === activeValue) {
            return null;
          }

          return (
            <DropdownItem
              toggle={false}
              key={value}
              className={classnames('dropdown-item-sm', styles['value-item'], {
                [styles['value-item__disabled']]: value === activeValue,
              })}
              onClick={handleClickItem(value)}
            >
              <IntlMessages id={getIntlId(value)} />
            </DropdownItem>
          );
        })}
      </DropdownMenu>
    </Dropdown>
  );
};
