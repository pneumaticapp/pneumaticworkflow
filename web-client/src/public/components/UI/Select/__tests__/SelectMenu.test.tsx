import * as React from 'react';
import { shallow } from 'enzyme';
import { DropdownItem } from 'reactstrap';

import { SelectMenu } from '../SelectMenu';
import { ETaskListSorting } from '../../../../types/tasks';

const mockProps = {
  activeValue: ETaskListSorting.DateDesc,
  values: Object.values(ETaskListSorting),
  onChange: jest.fn(),
};

describe('SelectMenu', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });
  it('корректно рендерится', () => {
    const wrapper = shallow(<SelectMenu {...mockProps}/>);

    expect(wrapper).toMatchSnapshot();
  });
  it('при клике на элемент вызывает onChange, если значение не активно', () => {
    const wrapper = shallow(<SelectMenu {...mockProps}/>);

    wrapper.find(DropdownItem).at(0).simulate('click');

    expect(mockProps.onChange).toHaveBeenCalledWith(ETaskListSorting.DateAsc);
  });
  it('при клике на элемент не вызывает onChange, если значение активно', () => {
    const wrapper = shallow(<SelectMenu {...mockProps}/>);

    wrapper.find(DropdownItem).at(1).simulate('click');

    expect(mockProps.onChange).not.toHaveBeenCalled();
  });
});
