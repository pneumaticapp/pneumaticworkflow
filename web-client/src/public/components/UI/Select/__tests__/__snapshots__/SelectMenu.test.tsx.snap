// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectMenu корректно рендерится 1`] = `
<Dropdown
  a11y={true}
  active={false}
  addonType={false}
  className="container dropdown-menu-right dropdown"
  direction="down"
  inNavbar={false}
  isOpen={false}
  nav={false}
  setActiveFromChild={false}
  toggle={[Function]}
>
  <DropdownToggle
    aria-haspopup={true}
    className="active-value"
    color="secondary"
    tag="button"
  >
    <ForwardRef
      id="sorting.date-desc"
    >
      <Component />
    </ForwardRef>
    <ExpandIcon
      className="expand-icon"
    />
  </DropdownToggle>
  <DropdownMenu
    className="dropdown-menu"
    flip={true}
    tag="div"
  >
    <DropdownItem
      className="dropdown-item-sm value-item"
      key="date-asc"
      onClick={[Function]}
      tag="button"
      toggle={false}
    >
      <ForwardRef
        id="sorting.date-asc"
      />
    </DropdownItem>
    <DropdownItem
      className="dropdown-item-sm value-item undefined"
      key="date-desc"
      onClick={[Function]}
      tag="button"
      toggle={false}
    >
      <ForwardRef
        id="sorting.date-desc"
      />
    </DropdownItem>
    <DropdownItem
      className="dropdown-item-sm value-item"
      key="overdue"
      onClick={[Function]}
      tag="button"
      toggle={false}
    >
      <ForwardRef
        id="sorting.overdue"
      />
    </DropdownItem>
  </DropdownMenu>
</Dropdown>
`;
